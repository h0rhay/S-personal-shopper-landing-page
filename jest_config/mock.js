export const mockTheme = {
  themeType: 'dark',
  breakpoints: {
    small: '(max-width: 812px)',
    medium: '(min-width: 813px)',
    large: '(min-width: 960px)',
    deviceSwitchSm: '(max-width: 896px)',
    deviceSwitchLg: '(min-width: 897px)',
  },
  typography: {
    mobile: {
      headerOne: {
        fontSize: '3rem',
      },
      headerTwo: {
        fontSize: '1.75rem',
      },
      displayHeading: {
        fontSize: 'calc(1.5rem * 2)',
      },
      itemHeading: {
        fontSize: '1.125rem',
      },
    },
    desktop: {
      headerOne: {
        fontSize: '4rem',
      },
      headerTwo: {
        fontSize: '2rem',
      },
      displayHeading: {
        fontSize: 'calc(4rem * 2)',
      },
      itemHeading: {
        fontSize: '1.125rem',
      },
    },
  },
  text: '#ffffff',
  primary: '#212121',
  primaryAccent: '#2E2E2E',
  accent: '#ffe256',
  shadow: 'rgba(0, 0, 0, 0.4)',
};
