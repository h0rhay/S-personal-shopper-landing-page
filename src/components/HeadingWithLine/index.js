import React from 'react';
import PropTypes from 'prop-types';
import styled, { css, keyframes } from 'styled-components';

const increaseAnimation = (height) => keyframes`
  0% {
    height: 0rem;
  }
  100% {
    height: ${height};
  }
`;

const HeadingLineWrap = styled.h1`
  line-height: 1;
  position: relative;
  ${(props) =>
    props.space
      ? css`
          padding-top: ${props.space}rem;
        `
      : null}
  ${(props) =>
    props.line
      ? css`
          &:before {
            position: absolute;
            content: '';
            top: ${props.position
              ? `
              ${props.position}rem
            `
              : `-20rem`};
            left: 1rem;
            height: ${props.lineHeight ? `${props.lineHeight}rem` : `16rem`};
            animation: ${props.lineHeight ? increaseAnimation(`${props.lineHeight}rem`) : increaseAnimation(`16rem`)} 3s;
            animation-iteration-count: 1;
            width: 1px;
            ${({ theme }) =>
              theme.themeType === 'dark'
                ? css`
                    background-color: ${theme.accent};
                  `
                : css`
                    background-color: ${theme.primaryAccent};
                  `}
            @media (max-width: 860px) {
              height: 8rem;
              top: -10rem;
            }

            @media (max-width: 860px) {
              display: none;
            }
          }
        `
      : null}
  @media(max-width:860px) {
    margin-top: 12rem;
  }
`;

const HeadingWithLine = ({ text, ...props }) => <HeadingLineWrap {...props}>{text}</HeadingLineWrap>;

HeadingWithLine.propTypes = {
  text: PropTypes.string,
};

export default HeadingWithLine;
