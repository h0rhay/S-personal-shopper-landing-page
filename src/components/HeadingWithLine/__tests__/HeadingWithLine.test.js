import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { mockTheme } from '../../../../jest_config/mock';
import HeadingWithLine from '../index';

const props = {
  text: 'This is a heading with a line',
  color: 'tomato',
  line: true,
  lineHeight: '10',
  space: '7',
  position: '-4',
};

const renderComponent = ({ theme }) =>
  render(
    <ThemeProvider theme={theme}>
      <HeadingWithLine {...props} />
    </ThemeProvider>
  );

describe('HeadingWithLine component', () => {
  it('should match screenshot', () => {
    const { container } = renderComponent({ theme: mockTheme, props });
    expect(container).toMatchSnapshot();
  });
});
