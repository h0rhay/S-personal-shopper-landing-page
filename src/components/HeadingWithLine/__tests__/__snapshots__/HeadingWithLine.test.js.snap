// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeadingWithLine component should match screenshot 1`] = `
<div>
  <h1
    class="sc-bdvvtL eUeKFG"
    color="tomato"
  >
    This is a heading with a line
  </h1>
</div>
`;
